### DISCUSSION TITLE
Why a new language and not a transpiler?

### DISCUSSION THREAD

**Discussion by grantjayy** [SaptakBhoumik/Peregrine#105] [General]
I love the idea behind this repo. 

Python is an amazing language but terribly slow. 

But why the need to create an entirely new language and not create a transpiler from python to C? Which it sounds like this language is doing in a nutshell. 

Also, will there be the ability to compile to Rust and not C?

**Comment by RocketMaker69** [comment]
I can only answer the last question: It would be better to compile to C, because Rust outruns C only in very basic programs, which means that if you have a big Peregrine program, C outruns Rust. But yes, there probably would be a way, but C is slightly better.

**Comment by SaptakBhoumik** [comment]
@grantjayy 
> why the need to create an entirely new language and not create a transpiler from python to C? 

If we make a transcompiler then we will be limited to python's syntax and limitation which we dont want. Also we want to add support for pointers and all those kind of stuff . If you want a transcompiler use cython

> Which it sounds like this language is doing in a nutshell.

In that sense v and nim are also transcompiler but they are not. Also peregrine has a secondary jit backend for prototyping purposes

> will there be the ability to compile to Rust and not C?

No. As stated by @RocketMaker69 It doesnt make any sense. Peregrine has it's own borrow checker so using a rust backend is as safe as a c backend
